client/src/ui/metricChart.tsx

```diff
@@ -0,0 +1,139 @@
+import Chart from 'react-apexcharts';
+import { useQuery } from "@tanstack/react-query";
+import { getManifestMetricSeries } from "api";
+import { 
+  Center,
+  Spinner, 
+  Box,
+  Text 
+} from "@liftedinit/ui";
+
+interface StatProps {
+  label: string;
+  type: any;
+  xtitle?: string,
+  ytitle?: string,
+  metric: string;
+  conversion?: Function;
+  from?: string;
+  to?: string;
+  fixedDecimals?: number;
+  intervalMs?: number;
+  maxDataPoints?: number;
+}
+
+interface ChartData {
+  series: { name: string; data: number[] }[];
+  options: any;
+}
+
+
+export function MetricChart({
+  label, 
+  type, 
+  metric, 
+  conversion, 
+  from, 
+  to, 
+  xtitle, 
+  ytitle, 
+  fixedDecimals,
+  intervalMs,
+  maxDataPoints,
+}: StatProps) {
+  const { data: queryData, isError, isLoading } = useQuery([metric + "series"], getManifestMetricSeries(metric,{from: from, to: to, intervalMs: intervalMs, maxDataPoints: maxDataPoints}));
+  let chartData: ChartData = {
+    series: [] ,
+    options: undefined
+  }
+  let categoriesData: (number|string)[] = []
+  let seriesData: number[] = []
+
+  if (isError || !queryData) {
+    categoriesData = [];
+    seriesData = [];
+  }
+
+  if (!isLoading && queryData) {
+    categoriesData = queryData[0]?.slice() || [];
+
+    if (conversion && queryData != null) {
+      seriesData = queryData[1]?.map(conversion)?.map((item: number) => Number((item).toFixed(fixedDecimals))) || [];
+    }
+    else if (!conversion && queryData != null ) {
+      seriesData = queryData[1]?.map((item: number) => Number((item).toFixed(fixedDecimals))) || [];
+    }
+    else {
+      seriesData = [];
+    }
+    
+    chartData = {
+      options: {
+        colors: ["#38C7B4"],
+        dataLabels: { enabled: false },
+        chart: { toolbar: { show: false } },
+        stroke: {
+          curve: 'smooth',
+        },
+        xaxis: {
+          title: { 
+            text: xtitle || "",
+            style: {
+              fontSize: "1.25em",
+              fontFamily: "Rubik, sans-serif",
+            },
+          },
+          type: 'datetime',
+          categories: categoriesData,
+          labels: {
+            overflow: 'scroll',
+            datetimeFormatter: {
+              year: 'yy',
+              month: 'MMM',
+              day: 'dd',
+              hour: 'HH:mm',
+            }
+        }
+        },
+        yaxis: {
+          title: { 
+            text: ytitle || "",
+            style: {
+              fontSize: "1.25em",
+              fontFamily: "Rubik, sans-serif",
+            },
+          },
+        }
+      },
+      series: [
+        {
+          name: label,
+          data: seriesData,
+        },
+      ],
+    };
+  }
+
+  return (
+    <>
+    {isLoading || isError ? (
```

Comment:
🤞 no errors! That's going to be spinning for a while. 😆

Reply:
Yes lol This is hacky to handle situations when Grafana API is unavailable. Will improve this in the next releases!